src/common/cluster/cluster.ts

```diff
@@ -340,15 +340,35 @@ export class Cluster implements ClusterModel, ClusterState {
     }
 
     if (this.disconnected || !this.accessible) {
-      await this.reconnect();
+      try {
+        this.broadcastConnectUpdate("Starting connection ...");
+        await this.reconnect();
+      } catch (error) {
+        this.broadcastConnectUpdate(`Failed to start connection: ${error}`, true);
+
+        return;
+      }
```

Comment:
We should do it in a way that doesn't require try-catches.

E.g. here it should be

```typescript

const reconnectResult = await this.reconnect();

if (reconnectResult) {
  // ...
} else {
  // ...
}

```

Adding try-catches is just invitation for this type of errors. Try-catch should be in the first place that might throw. Here it might be some shell command or `spawn` w.e.  

We have ~230 try-catch blocks in our code base. By doing it in this way, we would have ~90% less. And by doing that, this type of bugs would be impossible.